File: backend/app/schemas/activity.py
from datetime import datetime
from pydantic import BaseModel


class ActivityResponse(BaseModel):
    id: int
    user_id: int
    username: str | None
    activity_type: str
    movie_id: int | None
    movie_title: str | None
    reference_id: int | None
    created_at: datetime

    class Config:
        from_attributes = True


class ActivityListResponse(BaseModel):
    activities: list[ActivityResponse]
    total: int

File: frontend/src/api/activity.ts
import { fetchApi } from './auth';

export interface ActivityResponse {
  id: number;
  user_id: number;
  username: string | null;
  activity_type: string;
  movie_id: number | null;
  movie_title: string | null;
  reference_id: number | null;
  created_at: string;
}

export interface ActivityListResponse {
  activities: ActivityResponse[];
  total: number;
}

export async function getActivityFeed(skip = 0, limit = 20): Promise<ActivityListResponse> {
  return fetchApi<ActivityListResponse>(`/feed?skip=${skip}&limit=${limit}`);
}
